### PR TITLE
Attempting at enabling Gradle build scans.

### LINKED ISSUE #1526: Try Gradle build scan
Could be a good way to follow test status (instead of travis log)
https://gradle.com/build-scans

### PR DESCRIPTION
**Pull request by krmahadevan**
Fixes #1526  .

Closes #1526

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

Note:
The build still fails for OpenJDK7, but I think its not related to this change. We still have to fix the distribution that our OpenJDK7 uses.

Sample scan report : https://gradle.com/s/p6ique5m4tz2g

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @juherr @cbeust - Please help take a look

**Comment by juherr** [comment]
Example of result: https://scans.gradle.com/s/vm5gbe54vglpu

File: build.gradle [hunk 1]
```diff
@@ -26,12 +26,23 @@ buildscript {
...
     id "nebula.provided-base" version "3.1.0"
 }
 
+buildScan {
+    licenseAgreementUrl = 'https://gradle.com/terms-of-service'
+    licenseAgree = 'yes'
+    link 'GitHub', 'https://github.com/cbeust/testng/tree/master'
+    if (System.getenv('CI')) {
+        publishOnFailure()
+        tag 'CI'
```
Comment (juherr) [inline build.gradle]: Maybe we can add a tag for the jvm version

File: build.gradle [hunk 2]
```diff
@@ -26,12 +26,23 @@ buildscript {
...
     id "nebula.optional-base" version "3.1.0"
     id "nebula.provided-base" version "3.1.0"
 }
 
+buildScan {
+    licenseAgreementUrl = 'https://gradle.com/terms-of-service'
+    licenseAgree = 'yes'
+    link 'GitHub', 'https://github.com/cbeust/testng/tree/master'
+    if (System.getenv('CI')) {
+        publishOnFailure()
```
Comment (juherr) [inline build.gradle]: I think we can remove the test output now

File: build.gradle (hunk 1 above)
Comment (juherr) [reply build.gradle]: Forget this idea, it is already provided here https://scans.gradle.com/s/vm5gbe54vglpu#infrastructure

File: build.gradle (hunk 2 above)
Comment (krmahadevan) [reply build.gradle]: Fixed and pushed forcibly to the same commit. Please check

**Review by juherr — APPROVED** [review]
LGTM 👍 
Ping @cbeust

**Pull request merged**